bin/lsif_indexer.dart

```diff
@@ -28,15 +28,34 @@
 // Licensed under the BSD-2 Clause License: https://github.com/astashov/crossdart/blob/master/LICENSE
 
 import 'dart:io';
+import 'package:path/path.dart';
+
 import 'package:lsif_indexer/analyzer.dart';
+import 'package:lsif_indexer/src/arguments.dart';
+import 'package:lsif_indexer/src/emitter.dart';
+import 'package:lsif_indexer/src/util/path_extensions.dart';
 
-/// Generate LSIF information for the directory listed in the first [argument], or
+/// Generate LSIF information for the directory provided from the [arguments], or
 /// the current directory if not specified.
 ///
-/// Currently prints to standard out.
+/// The format of the output is based on the [arugments].
 void main(List<String> arguments) async {
-  // TODO: Allow specifying an output file.
-  await Analyzer(
-    arguments.isEmpty ? Directory.current.absolute.path : arguments.first,
-  ).analyzePackage();
+  final config = LsifDartArgumentParser().parse(arguments);
```

Comment:
naming nit: It's private to this package, so it doesn't need the disambiguation - it could just be ArgumentParser.  See https://journal.stuffwithstuff.com/2016/06/16/long-names-are-long/  (which I treat as a canonical part of Dart style, though it's actually separate from the style guide)

Reply:
Great blog post that I haven't read/seen before, thanks for sharing 👍